core/templates/dev/head/components/entity-creation-services/collection-creation-backend-api.service.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Service to notify about creation of collection and obtain
+ * collection_id.
+ */
+
+require('domain/utilities/url-interpolation.service.ts');
+require('services/site-analytics.service.ts');
+
+angular.module('oppia').factory('CollectionCreationBackendService', [
+  '$http', '$q',
+  function(
+      $http, $q) {
+    var _createCollection = function(successCallback, errorCallback) {
+      $http.post('/collection_editor_handler/create_new', {})
```

Comment:
@Arnesh07 Remove the empty parenthesis.

Reply:
I specifically mentioned this in the first point.
**So change to call to $http.post('/collection_editor_handler/create_new') and the test will start working since this is the request that you are trying to flush.**

Reply:
Yes I have done that. @Showtim3

Reply:
Still it doesn't work

Reply:
I tested this locally and is working for me. Can you push your changes?

Reply:
Do you want me to use the --no-verify flag?

Reply:
can you show my your code first? 
paste the code here where the api call is being made

Reply:
```
// Copyright 2019 The Oppia Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS-IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
/**
 * @fileoverview Service to notify about creation of collection and obtain
 * collection_id.
 */

angular.module('oppia').factory('CollectionCreationBackendService', [
  '$http', '$q',
  function(
      $http, $q) {
    var _createCollection = function(successCallback, errorCallback) {
      $http.post('/collection_editor_handler/create_new')
        .then(function(response) {
          if (successCallback) {
            successCallback(response.data);
          }
        }, function() {
          if (errorCallback) {
            errorCallback();
          }
        });
    };

    return {
      createCollection: function() {
        return $q(function(resolve, reject) {
          _createCollection(resolve, reject);
        });
      }
    };
  }
]);
```

Reply:
Paste the it block also

Reply:
```
it('should successfully create a new collection and obtain the collection ID',
    (done) => {
      var successHandler = jasmine.createSpy('success');
      var failHandler = jasmine.createSpy('fail');

      $httpBackend.expectPOST('/collection_editor_handler/create_new/').respond(
        200);
      CollectionCreationBackendService.createCollection().then(
        successHandler, failHandler);

      $httpBackend.flush();
      $rootScope.$digest();

      expect(successHandler).toHaveBeenCalled();
      expect(failHandler).not.toHaveBeenCalled();
      done();
    });
```

Reply:
and what's the error?

Reply:
```
Error: No pending request to flush !
	    at Function.$httpBackend.flush (third_party/static/angularjs-1.5.8/angular-mocks.js:1799:34)
	    at UserContext.eval (webpack:///./core/templates/dev/head/components/entity-creation-services/collection-creation-backend-api.service.spec.ts?:53:22)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:390:26)
	    at ProxyZoneSpec.onInvoke (webpack:///./node_modules/zone.js/dist/proxy.js?:128:39)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:389:52)
	    at Zone.run (webpack:///./node_modules/zone.js/dist/zone.js?:149:43)
	    at runInTestZone (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:173:34)
	    at UserContext.eval (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:189:20)
	    at <Jasmine>
	Error: Unsatisfied requests: POST /collection_editor_handler/create_new/
	    at Function.$httpBackend.verifyNoOutstandingExpectation (third_party/static/angularjs-1.5.8/angular-mocks.js:1832:13)
	    at UserContext.eval (webpack:///./core/templates/dev/head/components/entity-creation-services/collection-creation-backend-api.service.spec.ts?:45:22)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:390:26)
	    at ProxyZoneSpec.onInvoke (webpack:///./node_modules/zone.js/dist/proxy.js?:128:39)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:389:52)
	    at Zone.run (webpack:///./node_modules/zone.js/dist/zone.js?:149:43)
	    at runInTestZone (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:176:34)
	    at UserContext.eval (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:191:20)
	    at <Jasmine>
HeadlessChrome 78.0.3904 (Linux 0.0.0): Executed 1284 of 1438 (1 FAILED) (0 secs / 36.005 secs)
HeadlessChrome 78.0.3904 (Linux 0.0.0) Collection Creation backend service should successfully create a new collection and obtain the collection ID FAILED
	Error: No pending request to flush !
	    at Function.$httpBackend.flush (third_party/static/angularjs-1.5.8/angular-mocks.js:1799:34)
	    at UserContext.eval (webpack:///./core/templates/dev/head/components/entity-creation-services/collection-creation-backend-api.service.spec.ts?:53:22)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:390:26)
	    at ProxyZoneSpec.onInvoke (webpack:///./node_modules/zone.js/dist/proxy.js?:128:39)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:389:52)
	    at Zone.run (webpack:///./node_modules/zone.js/dist/zone.js?:149:43)
	    at runInTestZone (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:173:34)
	    at UserContext.eval (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:189:20)
	    at <Jasmine>
	Error: Unsatisfied requests: POST /collection_editor_handler/create_new/
	    at Function.$httpBackend.verifyNoOutstandingExpectation (third_party/static/angularjs-1.5.8/angular-mocks.js:1832:13)
	    at UserContext.eval (webpack:///./core/templates/dev/head/components/entity-creation-services/collection-creation-backend-api.service.spec.ts?:45:22)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:390:26)
	    at ProxyZoneSpec.onInvoke (webpack:///./node_modules/zone.js/dist/proxy.js?:128:39)
	    at ZoneDelegate.invoke (webpack:///./node_modules/zone.js/dist/zone.js?:389:52)
	    at Zone.run (webpack:///./node_modules/zone.js/dist/zone.js?:149:43)
	    at runInTestZone (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:176:34)
	    at UserContext.eval (webpack:///./node_modules/zone.js/dist/jasmine-patch.js?:191:20)
```

Reply:
Any updates? @Showtim3

Reply:
@Arnesh07 Once again ```'/collection_editor_handler/create_new/')``` and this ```expectPOST('/collection_editor_handler/create_new')``` are different.
Make them same and it'll work.

Reply:
Yes it works now, thanks!